src/System.Net.WebHeaderCollection/src/System/Net/WebHeaderCollection.cs

```diff
@@ -219,30 +402,158 @@ public override string ToString()
             return sb.ToString();
         }
 
+        public byte[] ToByteArray()
+        {
+            string tempString = this.ToString();
+            return System.Text.Encoding.ASCII.GetBytes(tempString);
+        }
+
         public WebHeaderCollection()
         {
-        }     
+        }
 
-        public int Count => _entriesList != null ? _entriesList.Count : 0;
+        public override int Count
+        {
+            get
+            {
+                return (_innerCollection == null ? 0 : _innerCollection.Count);
+            }
+        }
 
-        public string[] AllKeys => _allKeys ?? (_allKeys = ToArray(_entriesList));
+        public override string[] AllKeys
+        {
+            get
+            {
+                return InnerCollection.AllKeys;
+            }
+        }
 
-        IEnumerator IEnumerable.GetEnumerator()
+        public override IEnumerator GetEnumerator()
         {
-            EnsureInitialized();
-            return _entriesList.GetEnumerator();
+            return InnerCollection.Keys.GetEnumerator();
         }
+    }
 
-        private bool IsInitialized => _entriesList != null && _entriesDictionary != null;
+    internal class CaseInsensitiveAscii : IEqualityComparer, IComparer
```

Comment:
There's no built-in comparer we can use instead of this? e.g. CaseInsensitiveComparer.DefaultInvariant?